### PR TITLE
currency/manager: remove deadlock potential in concurrent operations

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Resolves intermittent deadlock observed in CI builds. [Example](https://github.com/thrasher-corp/gocryptotrader/actions/runs/9120973131/job/25079360158?pr=1082#step:10:279)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by thrasher- — APPROVED** [review]
Nice catch 👀

**Review by gloriousCode — APPROVED** [review]
tl;dr tACK! Thanks Shazbert. [Think I'll have to look at this as well](https://github.com/thrasher-corp/gocryptotrader/actions/runs/9121616981/job/25081078713?pr=1545#step:10:295) but much less severe

I wasn't sure of this solution initially since it was about `RLocks` and I know you can have multiple `RLocks`. However, in the order of things, once a `Lock` has been requested, all existing `RLocks` must be unlocked first and any `RLocks` requested after the `Lock` request (like `Get`), are queued for after the `Lock` is completed. I know I'm being 'splainy, but I had to check first as to why it was successful most of the time 😄

**Review by gloriousCode — APPROVED** [review]
ACK changes!
